internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1365 @@
+	VirtualSwitchConfigurationOverrideEnabled bool                                                       `tfschema:"virtual_switch_configuration_override_enabled"`
+}
+
+type HostNetworkIntentAdapterPropertyOverrideModel struct {
+	JumboPacket             string `tfschema:"jumbo_packet"`
+	NetworkDirect           string `tfschema:"network_direct"`
+	NetworkDirectTechnology string `tfschema:"network_direct_technology"`
+}
+
+type HostNetworkIntentQosPolicyOverrideModel struct {
+	BandWidthPercentageSMB         string `tfschema:"bandwidth_percentage_smb"`
+	PriorityValue8021ActionCluster string `tfschema:"priority_value8021_action_cluster"`
+	PriorityValue8021ActionSMB     string `tfschema:"priority_value8021_action_smb"`
+}
+
+type HostNetworkIntentVirtualSwitchConfigurationOverrideModel struct {
+	EnableIov              string `tfschema:"enable_iov"`
+	LoadBalancingAlgorithm string `tfschema:"load_balancing_algorithm"`
+}
+
+type HostNetworkStorageNetworkModel struct {
+	Name               string `tfschema:"name"`
+	NetworkAdapterName string `tfschema:"network_adapter_name"`
+	VlanId             string `tfschema:"vlan_id"`
+}
+
+type InfrastructureNetworkModel struct {
+	DhcpEnabled bool          `tfschema:"dhcp_enabled"`
+	SubnetMask  string        `tfschema:"subnet_mask"`
+	Gateway     string        `tfschema:"gateway"`
+	IpPool      []IpPoolModel `tfschema:"ip_pool"`
+	DnsServer   []string      `tfschema:"dns_server"`
+}
+
+type IpPoolModel struct {
+	StartingAddress string `tfschema:"starting_address"`
+	EndingAddress   string `tfschema:"ending_address"`
+}
+
+type OptionalServiceModel struct {
+	CustomLocation string `tfschema:"custom_location"`
+}
+
+type PhysicalNodeModel struct {
+	Name        string `tfschema:"name"`
+	Ipv4Address string `tfschema:"ipv4_address"`
+}
+
+type StorageModel struct {
+	ConfigurationMode string `tfschema:"configuration_mode"`
+}
+
+func (StackHCIDeploymentSettingResource) ModelObject() interface{} {
+	return &StackHCIDeploymentSettingModel{}
+}
+
+func (StackHCIDeploymentSettingResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"stack_hci_cluster_id": commonschema.ResourceIDReferenceRequiredForceNew(&deploymentsettings.ClusterId{}),
+
+		"arc_resource_ids": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: machines.ValidateMachineID,
+			},
+		},
+
+		"version": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$`),
+				"the version must be a set of numbers separated by dots: `10.0.0.1`",
+			),
+		},
+
+		"scale_unit": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"adou_path": {
```

Comment:
adou? what does this mean? google doesn't come up with anything

please make the property name more clear if it is an acronym,

Reply:
changed to `active_directory_organizational_unit_path`